pype/plugins/ftrack/publish/collect_ftrack_family.py

```diff
@@ -0,0 +1,62 @@
+"""
+Requires:
+    none
+
+Provides:
+    instance     -> families ([])
+"""
+import os
+import pyblish.api
+
+from pype.lib.plugin_tools import filter_profiles
+
+
+class CollectFtrackFamily(pyblish.api.InstancePlugin):
+    """
+        Adds explicitly 'ftrack' to families to upload instance to FTrack.
+
+        Uses selection by combination of hosts/families/tasks names via
+        profiles resolution.
+
+        Triggered everywhere, checks instance against configured
+    """
+    label = "Collect Ftrack Family"
+    order = pyblish.api.CollectorOrder + 0.4999
+
+    hosts = ["standalonepublisher"]
+
+    profiles = {
+        "hosts": ["standalonepublisher"],
+        "families": [],
+        "task_names": [],
+        "add_to_ftrack": True
+    }
+
+    def process(self, instance):
+        if not self.profiles:
+            self.log.warning("No profiles present for adding Ftrack family")
+            return
+
+        anatomy_data = instance.context.data["anatomyData"]
+        task_name = instance.data("task",
+                                  instance.context.data["task"])
+        host_name = anatomy_data.get("app",
+                                     os.environ["AVALON_APP"])
+        family = instance.data["family"]
+
+        filtering_criteria = {
+            "hosts": host_name,
+            "families": family,
+            "tasks": task_name
+        }
+        profile = filter_profiles(self.profiles, filtering_criteria)
+
```

Comment:
I would add one similar condition like at the top `if not profile:` which means that none of profiles match current context which should be definetelly logged.

Reply:
Changed logging of result a bit.

Reply:
Still missing information that none of profiles match current context.

Reply:
I dont think its necessary, its is already logging quite a bit in `filter_profiles` function. I think that not finding matching profile is perfectly valid state not needing a log or warning.
And its instance plugin, so it might keep repeating "This instance doesnt match to any profile...This instance doesn't match...."

Reply:
> its is already logging quite a bit in filter_profiles

But that is not displayed in Publish UI.

> not needing a log or warning.

It does not have to be a warning but debug. It's for us, not artist. Artist will never read them...

Reply:
But if you'll pass logger to `filter_profiles` it's enough.

Reply:
Ok.